File: SingleElementInASortedArray.cpp
#include<bits/stdc++.h>
using namespace std;
// leetcode

int singleNonDuplicate(vector<int> & nums){
    int low = 0,high = nums.size()-1;
    
    int mid;
    while(low <= high){
        mid = low + (high - low)/2;
        if (mid % 2==1) mid--;//check if it is even index to check for a pair

        if (nums[mid] != nums[mid + 1])high = mid;//if the even index is not matching a pair then missing element is on the left side
        else low = mid + 2;
    }
    return nums[low];

}

File: single-element-in-a-sorted-array/single-element-in-a-sorted-array.java
class Solution {
    public int singleNonDuplicate(int[] nums) {
        int low =0,high= nums.length-1;
        if (nums.length==1)return nums[0];
        while(low<high){
            int mid = low + (high-low)/2;
            if (mid %2==0 && nums[mid+1] == nums[mid] || (mid%2==1 && nums[mid] == nums[mid-1])){
                //the single ele is on the right
                low = mid+1;
            }
            else high= mid;
        }
        return nums[low];
    }
}